### PR TITLE
Big Auth / Integrations Docs Update

### PR DESCRIPTION
**Pull request by vincanger**
### Description

Did a big Doc update to the following sections:
- Examples section
- Social Auth Providers,
- re-organized the Sidebar sections 
- created more in-depth Auth Integrations sections
image

### Select what type of change this PR introduces:

1. [x] **Just code/docs improvement** (no functional change). 
2. [ ] **Bug fix** (non-breaking change which fixes an issue).
3. [ ] **New feature** (non-breaking change which adds functionality).
4. [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

### CODE REVIEW

File: web/docs/integrations/github.md [hunk 1]
```diff
@@ -6,9 +6,60 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
 # GitHub Integrations
```
Comment (infomiho) [inline web/docs/integrations/github.md:7]: Maybe change the title from "GitHub Integrations" to just "GitHub". (same goes for Google page)

Screenshot 2023-01-30 at 16 04 02

File: web/docs/language/features.md [hunk 2]
```diff
@@ -1171,15 +1171,19 @@ entity SocialLogin {=psl
...
+When a user signs in for the first time, Wasp assigns generated values to the `username` and `password` fields of the `userEntity` by default (e.g. `username: nice-blue-horse-14357`), so make sure to include these in your `userEntity` declaration even if you're only using a Social Login provider. If you'd like to change this behavior, these values can be overridden as described below.
 :::
 
+:::tip UI Helpers and Overrides
+Wasp provides you with a few UI helper components and functions to make implementing social login easier. See [here](features#ui-helpers-for-social-login-providers) for more details.
+
+It is also posslbe to [override the default](features#overrides-for-social-login-providers) login behaviors that Wasp provides for you. This allows you to create custom setups, such as allowing Users to define a username rather than the default random username assigned by Wasp on initial Login.
+:::
 
 #### Google
```
Comment (infomiho) [inline web/docs/language/features.md:1188]: Maybe make it a bigger title with `###` and maybe link to the guide below (e.g. "check out our guide on setting up Google auth")

Screenshot 2023-01-30 at 16 10 22

Comment (vincanger) [reply web/docs/language/features.md]: ok. I made some more improvements and changes to remove redundant information

**Comment by Martinsos** [comment]
idea: @vincanger maybe we should put Example at the top level, next to Tutorials, Guides and similar? They are quite important I think and they are technically not guides.

**Review by infomiho — APPROVED** [review]
Great update 👏 

@shayneczyzewski you'll maybe have to fix your deployment docs changes since this moves files around a bit

**Comment by shayneczyzewski** [comment]
> Great update 👏
> 
> @shayneczyzewski you'll maybe have to fix your deployment docs changes since this moves files around a bit

Thanks for the heads up @infomiho. I don't think there is overlap but appreciate you noting just in case!

**Pull request merged**